processing/src/main/java/org/apache/druid/query/filter/RangeFilter.java

```diff
@@ -812,6 +826,25 @@ public String toString()
     }
   }
 
+  /**
+   * if casting to a LONG typed ARRAY, first cast to a DOUBLE typed ARRAY and then {@link Math#ceil(double)} on the
```

Comment:
Would be good to say that the intent of this method is to be like `castTo(inputType).asArray()`, but with one difference: when the target is long, and the inputs are decimals, we ceil them instead of flooring them.